homeassistant/components/sensibo/climate.py

```diff
@@ -102,244 +84,242 @@ async def async_setup_entry(
-            await self._async_set_ac_state_property("on", state != HVAC_MODE_OFF, True)
-
-        if state in (STATE_ON, HVAC_MODE_OFF):
-            self._external_state = None
-        else:
-            self._external_state = state
-
-    async def async_update(self) -> None:
-        """Retrieve latest state."""
-        try:
-            async with async_timeout.timeout(TIMEOUT):
-                data = await self._client.async_get_device(self._id, _FETCH_FIELDS)
-        except (
-            aiohttp.client_exceptions.ClientError,
-            asyncio.TimeoutError,
-            SensiboError,
-        ) as err:
-            if self._failed_update:
-                _LOGGER.warning(
-                    "Failed to update data for device '%s' from Sensibo servers with error %s",
-                    self._attr_name,
-                    err,
-                )
-                self._attr_available = False
-                self.async_write_ha_state()
-                return
-
-            _LOGGER.debug("First failed update data for device '%s'", self._attr_name)
-            self._failed_update = True
-            return
-
-        if self.temperature_unit == self.hass.config.units.temperature_unit:
-            self._attr_target_temperature_step = 1
-        else:
-            self._attr_target_temperature_step = None
-
-        self._failed_update = False
-        self._do_update(data)
+        result = await self._async_set_ac_state_property("on", False)
+        if result:
+            self.coordinator.data[self.unique_id]["on"] = False
+            self.async_write_ha_state()
 
     async def _async_set_ac_state_property(
-        self, name, value, assumed_state=False
-    ) -> None:
+        self, name: str, value: Any, assumed_state: bool = False
+    ) -> bool:
         """Set AC state."""
+        result = {}
         try:
             async with async_timeout.timeout(TIMEOUT):
-                await self._client.async_set_ac_state_property(
-                    self._id, name, value, self._ac_states, assumed_state
+                result = await self._client.async_set_ac_state_property(
+                    self.unique_id,
+                    name,
+                    value,
+                    self.coordinator.data[self.unique_id]["ac_states"],
+                    assumed_state,
                 )
         except (
-            aiohttp.client_exceptions.ClientError,
+            ClientConnectionError,
             asyncio.TimeoutError,
             SensiboError,
         ) as err:
-            self._attr_available = False
-            self.async_write_ha_state()
-            raise Exception(
-                f"Failed to set AC state for device {self._attr_name} to Sensibo servers"
+            raise HomeAssistantError(
+                f"Failed to set AC state for device {self.name} to Sensibo servers: {err}"
             ) from err
+        if result["status"] == "Success":
+            return True
+        return False
```

Comment:
When will it not return success and also not raise an exception?

Reply:
If the local device is offline e.g. unplugged or something like such. It's not handled in the upstream package.
Not sure if @andrey-git is still active but seems as he hasn't done anything since 2018 and therefore probably this package should be rewritten later.

Reply:
When the device is offline we should probably set the entity as unavailable. That will also guard against calling services on the entity at that time.

Reply:
It's also included but since polling is "only" per 60 seconds there might be a call to the api while HA has not updated it as unavailable yet
https://github.com/home-assistant/core/blob/285c0b8b1b8c1bcb7286937e8f68e22113dee387/homeassistant/components/sensibo/climate.py#L224

Reply:
Should we raise HomeAssistantError when the call failed due to device being offline?